tests/e2e/testsuites/testsuites.go

```diff
@@ -544,15 +546,22 @@ func podLogs(client clientset.Interface, name, namespace string) ([]byte, error)
 }
 
 // Method to get cloudInstanceId from Node Labels
-func GetCloudInstanceIdFromNodeLabels(client clientset.Interface) (string, error) {
+func GetCloudInstanceIdFromNodeSpec(client clientset.Interface) (string, error) {
 	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return "", fmt.Errorf("error getting cloudInstanceId: %+v", err)
 	}
 
-	if cloudInstanceId, ok := nodes.Items[0].ObjectMeta.Labels[powervscloud.CloudInstanceIDLabel]; ok {
+	// ProviderID format: ibmpowervs://<region>/<zone>/<service_instance_id>/<powervs_machine_id>
+	providerId := nodes.Items[0].Spec.ProviderID
+	if providerId != "" {
+		klog.Infof("PROVIDER-ID: %s", providerId)
+		data := strings.Split(providerId, "/")
+		if len(data) != powervscloud.ProviderIDValidLength {
+			return "", fmt.Errorf("invalid ProviderID format - %v, expected format - ibmpowervs://<region>/<zone>/<service_instance_id>/<powervs_machine_id>", providerId)
+		}
+		cloudInstanceId := data[4]
```

Comment:
I see this code is repeating, wondering if we can have this in some common place to reuse.

Reply:
code refactored.